MantidQt/CustomInterfaces/src/Poldi/PoldiPresenter.cpp

```diff
@@ -0,0 +1,106 @@
+#include "MantidQtCustomInterfaces/Poldi/PoldiPresenter.h"
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidQtCustomInterfaces/Poldi/IPoldiView.h"
+#include "MantidQtMantidWidgets/DataProcessorUI/DataProcessorCommand.h"
+#include <Poco/ActiveResult.h>
+#include <qapplication.h>
+
+using namespace Mantid::API;
+
+namespace MantidQt {
+namespace CustomInterfaces {
+
+/** Constructor
+* @param view : [input] The view this presenter handles
+* @param presenter : [input] The DataProcessor presenter that will tell us when
+* the ADS changes
+*/
+PoldiPresenter::PoldiPresenter(
+    IPoldiView *view, boost::shared_ptr<DataProcessorPresenter> presenter)
+    : m_view(view), m_presenter(presenter) {
+
+  // This is very important, it means that our table presenter will use this
+  // presenter to communicate changes in the ADS
+  // Without this the interface will crash
+  m_presenter->accept(this);
+
+  // Set up the instrument selectors
+  std::vector<std::string> instruments = {"POLDI"};
+  m_presenter->setInstrumentList(instruments, "POLDI");
+}
+
+/** Default destructor */
+PoldiPresenter::~PoldiPresenter() {}
+
+/** This method is used by the view to tell this presenter that something has
+ * happened
+ * @param flag : [input] A flag indicating what happened
+ */
+void PoldiPresenter::notify(IPoldiPresenter::Flag flag) {
+
+  switch (flag) {
+  case IPoldiPresenter::LoadDemoFlag:
+    loadDemoTable();
+    break;
+  }
+}
+
+/** This method is used by the DataProcessor presenter to tell this presenter
+* that something has changed in the ADS. We don't need to do anything, as in
+* this interface, but we need to have this flag defined.
+* @param flag : [input] A flag indicating what happened (currently only one
+* possible flag)
+*/
+void PoldiPresenter::notify(WorkspaceReceiver::Flag flag) {
+
+  switch (flag) {
+  case WorkspaceReceiver::ADSChangedFlag:
+    // Do nothing
+    break;
+  }
+}
+
+/** Slot to load a demo table following user example in PoldiDataAnalysis
+ * documentation */
+void PoldiPresenter::loadDemoTable() {
```

Comment:
Does it make sense to consider this as a `toy` demo? As you've pointed it at Poldi and have the help of @MichaelWedel. Would it make sense to extend the working functionality of this? Not sure I know the answer to this.